stock_move_backdating/models/stock_move.py

```diff
@@ -21,5 +21,11 @@ def action_done(self):
                 move.date = operation.operation_id.date
                 if move.quant_ids:
                     move.quant_ids.sudo().write({'in_date': move.date})
-
+        # set date_done as the youngest date among the pack operations
+        operations = self.mapped(
+            'linked_move_operation_ids.operation_id')
+        dates = operations.mapped('date')
+        pickings = operations.mapped('picking_id')
+        pickings = pickings.filtered(lambda r: r.state == 'done')
+        pickings.write({'date_done': max(dates)})

```

Comment:
In questo modo se non sbaglio stai prendendo tutti i picking su cui ```action_done``` viene chiamato, prendi tutte le move di tutti i picking, guardi quella con data + recente e scrivi questa data uguale per tutti i picking.

Invece dovresti prendere la data + recente per ogni picking